homeassistant/components/rfxtrx.py

```diff
@@ -191,6 +214,78 @@ def get_rfx_object(packetid):
     return obj
 
 
+def get_pt2262_deviceid(device_id, nb_data_bits):
+    """Extract and return the address bits from a Lighting4/PT2262 packet."""
+    import binascii
+    try:
+        data = bytearray.fromhex(device_id)
+    except ValueError:
+        return None
+
+    mask = 0xFF & ~((1 << nb_data_bits) - 1)
+
+    data[len(data)-1] &= mask
+
+    return binascii.hexlify(data)
+
+
+def get_pt2262_cmd(device_id, data_bits):
+    """Extract and return the data bits from a Lighting4/PT2262 packet."""
+    try:
+        data = bytearray.fromhex(device_id)
+    except ValueError:
+        return None
+
+    mask = 0xFF & ((1 << data_bits) - 1)
+
+    return hex(data[-1] & mask)
+
+
+# pylint: disable=unused-variable
+def get_pt2262_device(device_id):
+    """Look for the device which id matches the given device_id parameter."""
+    for dev_id, device in RFX_DEVICES.items():
+        try:
+            if (device.is_pt2262 and
+                    device.masked_id == get_pt2262_deviceid(
+                        device_id,
+                        device.data_bits)):
+                _LOGGER.info("rfxtrx: found matching device %s for %s",
+                             device_id,
+                             get_pt2262_deviceid(device_id, device.data_bits))
+                return device
+        except AttributeError:
+            continue
+    return None
+
+
+# pylint: disable=unused-variable
+def find_possible_pt2262_device(device_id):
+    """Look for the device which id matches the given device_id parameter."""
+    for dev_id, device in RFX_DEVICES.items():
+        if len(dev_id) == len(device_id):
+            size = None
+            for i in range(0, len(dev_id)):
+                if dev_id[i] != device_id[i]:
+                    break
+                size = i
+
+            if size is not None:
+                size = len(dev_id) - size - 1
+                _LOGGER.info("rfxtrx: found possible device %s for %s "
+                             "with the following configuration:\n"
+                             "data_bits=%d\n"
+                             "command_on=0x%s\n"
+                             "command_off=0x%s\n",
+                             device_id,
+                             dev_id,
+                             size * 4,
+                             dev_id[-size:], device_id[-size:])
+                return device
```

Comment:
Same thing as prior comment. This method does not init a device object but returns an existing one, therefore with a its "hass" property already set.